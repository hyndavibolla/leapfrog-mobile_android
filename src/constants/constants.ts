export const pushNotificationTypes = {
  USER_INACTIVE: 'USER_INACTIVE',
  MISSION_COMPLETE: 'MISSION_COMPLETE',
  BALANCE_NEW_POINTS: 'BALANCE_NEW_POINTS',
  BALANCE_EXPIRING_POINTS: 'BALANCE_EXPIRING_POINTS',
  BALANCE_UNUSED_POINTS: 'BALANCE_UNUSED_POINTS',
  MARKETING: 'MARKETING',
  USER_BIRTHDAY: 'USER_BIRTHDAY'
};

export const RedemptionMethods = {
  IN_STORE_ONLY: 'In-Store only',
  ONLINE_ONLY: 'Online only',
  IN_STORE_OR_ONLINE_ONLY: 'In-Store and Online only',
  IN_RESTAURANT_ONLY: 'In Restaurant only',
  APP_ONLY: 'App only'
};

export const GROCERY_AND_DELIVERY_CATEGORY = 'Grocery & Delivery';

export const TutorialFrom = {
  TUTORIAL_REMINDER: 'tutorial_reminder',
  TUTORIAL_BANNER: 'tutorial_banner',
  PROFILE: 'profile'
};
