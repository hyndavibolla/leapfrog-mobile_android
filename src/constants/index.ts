export * from './env';
export * from './routes';
export * from './eventTracking';
export * from './styles';
export * from './icons';
export * from './constants';
