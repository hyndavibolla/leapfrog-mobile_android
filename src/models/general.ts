import { Dispatch } from 'react';
import Auth0 from 'react-native-auth0';

import { ApiService } from '_services/ApiService';
import { AppService } from '_services/AppService';
import { CCPAService } from '_services/CCPAService';
import { ICrashlyticsService } from '_services/CrashlyticsService';
import { EventTrackerService } from '_services/EventTrackerService';
import { IHttpService } from '_services/HttpService';
import { NativeHelperService } from '_services/NativeHelperService';
import { Logger } from '_services/Logger';
import { IPerformanceTrackingService } from '_services/PerformanceTrackingService';
import { RaiseService } from '_services/RaiseService';
import { RemoteConfigService } from '_services/RemoteConfigService';
import { ICardLinkState } from '_state_mgmt/cardLink/state';
import { ICoreState } from '_state_mgmt/core/state';
import { IGameState } from '_state_mgmt/game/state';
import { IGiftCardState } from '_state_mgmt/giftCard/state';
import { IMessagesState } from '_state_mgmt/messages/state';
import { IMissionState } from '_state_mgmt/mission/state';
import { IRewardState } from '_state_mgmt/reward/state';
import { ISurveyState } from '_state_mgmt/survey/state';
import { IUserState } from '_state_mgmt/user/state';

export interface IAction<T = string, P = any> {
  type: T;
  payload: P;
}

export type IReducer<S = any, A = any> = (state: S, action: IAction<A>) => S;

export interface Deps {
  apiService: ApiService;
  appService: AppService;
  crashlyticsService: ICrashlyticsService;
  httpService: IHttpService;
  performanceTrackingService: IPerformanceTrackingService;
  raiseService: RaiseService;
  nativeHelperService: NativeHelperService;
  stateSnapshot: { get: () => IGlobalState; set: (state: IGlobalState) => void };
  logger: Logger;
  eventTrackerService: EventTrackerService;
  ccpaService: CCPAService;
  remoteConfigService: RemoteConfigService;
  authService: Auth0;
}
export type IGlobalState = {
  core: ICoreState;
  user: IUserState;
  game: IGameState;
  mission: IMissionState;
  reward: IRewardState;
  survey: ISurveyState;
  messages: IMessagesState;
  cardLink: ICardLinkState;
  giftCard: IGiftCardState;
};

export interface IGlobalContext {
  state: IGlobalState;
  dispatch: Dispatch<IAction<any, any>>;
  deps: Deps;
}

export enum HTTP_METHOD {
  GET = 'GET',
  POST = 'POST',
  PUT = 'PUT',
  PATCH = 'PATCH',
  DELETE = 'DELETE'
}

export interface ITokenSet {
  accessToken: string;
  refreshToken: string;
  accessTokenExpiryTime: string;
  sywToken?: string;
  /**
   * @deprecated deviceId does not belong to ITokenSet anymore.
   */
  deviceId?: string;
  scope?: string;
  expiresIn?: number;
  tokenType?: string;
}

export enum FeatureFlag {
  API_OVERRIDE = 'API_OVERRIDE',
  TEST_ONLY = 'TEST_ONLY',
  ENV_BANNER = 'ENV_BANNER',
  BUTTON_DEBUG = 'BUTTON_DEBUG',
  LOG_LEVEL_INFO = 'LOG_LEVEL_INFO',
  LOG_LEVEL_DEBUG = 'LOG_LEVEL_DEBUG',
  LOG_LEVEL_WARN = 'LOG_LEVEL_WARN',
  LOG_LEVEL_ERROR = 'LOG_LEVEL_ERROR',
  LOG_VIEW = 'LOG_VIEW',
  STREAK_REDEEM_CARD = 'STREAK_REDEEM_CARD',
  SAILTHRU_TRACKING = 'SAILTHRU_TRACKING',
  SURVEY = 'SURVEY',
  PREDICTIVE_MAP_SEARCH = 'PREDICTIVE_MAP_SEARCH',
  MISSIONS_SEARCH = 'MISSIONS_SEARCH',
  SEE_ALL_TRANSACTIONS = 'SEE_ALL_TRANSACTIONS',
  WALLET_YOUR_GC = 'WALLET_YOUR_GC'
}

export type DateLike = Date | string | number;

export type HTMLString = string;

export interface IGameLevelUI {
  updatedAt: DateLike;
  level: number;
}

export interface ICCPASetting {
  allow: boolean;
}

export interface IOnboarding {
  seenOnboarding: boolean;
}

export enum ButtonCreativeType {
  HERO = 'hero',
  CAROUSEL = 'carousel',
  LIST = 'list',
  GRID = 'grid',
  DETAIL = 'detail',
  OTHER = 'other'
}

export interface ISearchHistoryItem {
  keyword: string;
  lastUpdatedAt: DateLike;
}

export enum TooltipKey {
  EARN = 'earn',
  REWARDS = 'rewards',
  MISSIONS = 'missions',
  STREAK = 'streak'
}

export interface IApiOverrideRule {
  key: string;
  description?: string;
  matcher: string; // reg exp
  status: number;
  response: any;
}

export namespace RemoteConfigModel {
  export interface IAppVersion {
    suggested: number;
    required: number;
  }

  export interface IEarnSection {
    name: string;
    order: number;
    visible: boolean;
    title: string;
    description: string;
    seeAllButton: boolean;
  }
}

export enum RUNTIME_ENV {
  LIVE = 'live',
  DEV_TEST = 'test'
}

export interface IMeasure {
  x: number;
  y: number;
  width: number;
  height: number;
}

export interface IExpirePointsBannerDataSet {
  lastPointsDate: DateLike;
  lastAcceptedDate: DateLike;
}

export interface ISailthruMessage {
  createdAt: string;
  custom: {
    [k: string]: string;
  };
  htmlText: string;
  id: string;
  isRead: boolean;
  text: string;
  title: string;
  type: string;
  cardImageUrl?: string;
  share: boolean;
}

export interface ITutorialStatus {
  isBannerWatched: boolean;
  date: string;
}
