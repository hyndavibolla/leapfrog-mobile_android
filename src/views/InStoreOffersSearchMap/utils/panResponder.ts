const swipeConfig = {
  velocityThreshold: 0.3,
  directionalOffsetThreshold: 80,
  gestureIsClickThreshold: 5
};

const swipeDirections = {
  swipeUp: 'SWIPE_UP',
  swipeDown: 'SWIPE_DOWN'
};

export const handleShouldSetPanResponder = ({ nativeEvent: { touches } }, gestureState) => {
  return touches.length === 1 && !gestureIsClick(gestureState);
};

export const gestureIsClick = ({ dx, dy }) => {
  return Math.abs(dx) < swipeConfig.gestureIsClickThreshold && Math.abs(dy) < swipeConfig.gestureIsClickThreshold;
};

export const handlePanResponderEnd = (gestureState, onSwipeUp, onSwipeDown) => {
  const swipeDirection = getSwipeDirection(gestureState);
  triggerSwipeHandlers(swipeDirection, onSwipeUp, onSwipeDown);
};

export const triggerSwipeHandlers = (swipeDirection, onSwipeUp, onSwipeDown) => {
  const { swipeUp, swipeDown } = swipeDirections;
  switch (swipeDirection) {
    case swipeUp:
      onSwipeUp();
      break;
    case swipeDown:
      onSwipeDown();
      break;
  }
};

export const getSwipeDirection = gestureState => {
  const { swipeUp, swipeDown } = swipeDirections;
  if (isValidVerticalSwipe(gestureState)) {
    return gestureState.dy > 0 ? swipeDown : swipeUp;
  }
  return null;
};

export const isValidVerticalSwipe = ({ vy, dx }) => {
  const { velocityThreshold, directionalOffsetThreshold } = swipeConfig;
  return isValidSwipe(vy, velocityThreshold, dx, directionalOffsetThreshold);
};

export const isValidSwipe = (velocity, velocityThreshold, directionalOffset, directionalOffsetThreshold) => {
  return Math.abs(velocity) > velocityThreshold && Math.abs(directionalOffset) < directionalOffsetThreshold;
};
