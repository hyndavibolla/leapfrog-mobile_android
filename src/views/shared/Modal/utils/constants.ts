export enum ModalType {
  CENTER = 'center',
  BOTTOM = 'bottom'
}

export enum ModalSize {
  SMALL = 'small',
  MEDIUM = 'medium',
  EXTRA_LARGE = 'extralarge',
  DYNAMIC = 'dynamic',
  FULL_SCREEN = 'fullscreen'
}

export const ModalSizeToHeightMap = {
  [ModalSize.SMALL]: 40,
  [ModalSize.MEDIUM]: 50,
  [ModalSize.EXTRA_LARGE]: 80,
  [ModalSize.FULL_SCREEN]: 100
};
