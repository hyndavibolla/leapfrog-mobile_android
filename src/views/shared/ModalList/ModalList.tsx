import React, { memo, useContext, useCallback, useState, useEffect } from 'react';

import { Modal } from '../Modal';
import { GlobalContext } from '../../../state-mgmt/GlobalState';
import { actions } from '../../../state-mgmt/core/actions';
import { ENV } from '../../../constants';

export interface Props {}

export const ModalList = ({}: Props) => {
  const { state, dispatch } = useContext(GlobalContext);
  const [visible, setVisible] = useState<boolean>(false);

  const [item] = state.core.modalList;

  const onPressOutside = useCallback(() => {
    if (item.props.onPressOutside) item.props.onPressOutside();
    setVisible(false);
  }, [item]);

  const onClose = useCallback(() => {
    if (item.props.onClose) item.props.onClose();
    setTimeout(() => dispatch(actions.removeModal(item.key)), ENV.MODAL_PADDING_MS);
  }, [item, dispatch]);

  useEffect(() => {
    if (item) setVisible(true);
  }, [item]);

  return !item ? null : <Modal key={item.key} visible={visible} {...item.props} onPressOutside={onPressOutside} onClose={onClose} />;
};

export default memo(ModalList);
