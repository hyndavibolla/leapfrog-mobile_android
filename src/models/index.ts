import * as GeneralModel from './general';
import * as UserModel from './user';
import * as ActivityModel from './activity';
import * as OfferModel from './offer';
import * as GameModel from './game';
import * as MissionModel from './mission';
import * as RewardModel from './reward';
import * as StreakModel from './streak';
import * as PromptModel from './prompt';
import * as CardLink from './cardLink';

export { GeneralModel, UserModel, ActivityModel, OfferModel, GameModel, MissionModel, RewardModel, StreakModel, PromptModel, CardLink };
