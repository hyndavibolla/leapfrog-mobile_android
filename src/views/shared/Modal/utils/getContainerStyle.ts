import { memorize } from '../../../../utils/memorize';
import { styles } from '../styles';
import { ModalSize, ModalType, ModalSizeToHeightMap } from './constants';

interface props {
  height: number;
  modalHeight: number;
  size: ModalSize;
  type: ModalType;
}

const getContainerStyle = ({ height, modalHeight, size, type }: props) => {
  switch (type) {
    case ModalType.BOTTOM:
      if (size === ModalSize.FULL_SCREEN) return [styles.containerBottom, styles.containerBottomFullScreen];
      else if (size !== ModalSize.DYNAMIC) return [styles.containerBottom, { maxHeight: (ModalSizeToHeightMap[size] * height) / 100 }];
      return [styles.containerBottom];
    case ModalType.CENTER:
      if (size === ModalSize.DYNAMIC) return [styles.containerCenter, styles.containerCenterDynamic, { bottom: (height - modalHeight) / 2 }];
      const center = (ModalSizeToHeightMap[size] * height) / 100;
      return [styles.containerCenter, { height: center, bottom: (height - center) / 2 }];
  }
};

export default memorize(getContainerStyle);
