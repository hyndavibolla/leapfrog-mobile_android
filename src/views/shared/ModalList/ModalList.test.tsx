import React from 'react';
import { act } from 'react-test-renderer';
import { fireEvent } from '@testing-library/react-native';

import ModalList, { Props } from './ModalList';
import { wait } from '../../../utils/wait';
import { ENV } from '../../../constants';
import { renderWithGlobalContext } from '../../../test-utils/renderWithGlobalContext';
import { getInitialState } from '../../../state-mgmt/GlobalState';
import { IGlobalState } from '../../../models/general';

describe('ModalList', () => {
  let props: Props;
  let initialState: IGlobalState;

  beforeEach(() => {
    initialState = getInitialState();
    props = {};
  });

  it('should render all items one at a time', async () => {
    const { toJSON, getByTestId } = renderWithGlobalContext(<ModalList {...props} />, undefined, {
      ...initialState,
      core: {
        ...initialState.core,
        modalList: [
          { key: '1', props: { children: 'some text' } },
          { key: '2', props: { children: 'some other text' } }
        ]
      }
    });
    expect(toJSON()).toMatchSnapshot();
    fireEvent.press(getByTestId('modal-backdrop'));
    await act(() => wait(ENV.MODAL_PADDING_MS));
    expect(toJSON()).toMatchSnapshot();
    fireEvent.press(getByTestId('modal-backdrop'));
    await act(() => wait(ENV.MODAL_PADDING_MS));
    expect(toJSON()).toMatchSnapshot();
  });

  it('should call on close and on press outside callbacks', async () => {
    const onClose = jest.fn();
    const onPressOutside = jest.fn();
    const { getByTestId } = renderWithGlobalContext(<ModalList {...props} />, undefined, {
      ...initialState,
      core: {
        ...initialState.core,
        modalList: [{ key: '1', props: { children: 'some text', onClose, onPressOutside } }]
      }
    });
    fireEvent.press(getByTestId('modal-backdrop'));
    await act(() => wait(ENV.MODAL_PADDING_MS));
    expect(onPressOutside).toBeCalled();
    expect(onClose).toBeCalled();
  });
});
