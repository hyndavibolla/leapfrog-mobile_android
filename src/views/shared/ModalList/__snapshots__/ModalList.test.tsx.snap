// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalList should render all items one at a time 1`] = `
<Modal
  accessibilityLabel="modal-container"
  accessible={true}
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  statusBarTranslucent={true}
  testID="modal-container"
  transparent={true}
  visible={true}
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "bottom": 0,
          "position": "absolute",
          "zIndex": 9999,
        },
        Object {
          "borderTopLeftRadius": 14,
          "borderTopRightRadius": 14,
          "bottom": 0,
          "left": 0,
          "paddingBottom": 60,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 60,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    some text
  </View>
  <div
    animation="fadeIn"
    delay={400}
    duration={400}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(38, 44, 47, 0.45)",
          "height": "100%",
          "left": 0,
          "top": 0,
          "width": "100%",
        },
        undefined,
      ]
    }
    useNativeDriver={true}
  >
    <View
      accessibilityLabel="modal-backdrop"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="modal-backdrop"
    >
      <View
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </View>
  </div>
</Modal>
`;

exports[`ModalList should render all items one at a time 2`] = `
<Modal
  accessibilityLabel="modal-container"
  accessible={true}
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  statusBarTranslucent={true}
  testID="modal-container"
  transparent={true}
  visible={true}
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "bottom": 0,
          "position": "absolute",
          "zIndex": 9999,
        },
        Object {
          "borderTopLeftRadius": 14,
          "borderTopRightRadius": 14,
          "bottom": 0,
          "left": 0,
          "paddingBottom": 60,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 60,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    some other text
  </View>
  <div
    animation="fadeIn"
    delay={400}
    duration={400}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(38, 44, 47, 0.45)",
          "height": "100%",
          "left": 0,
          "top": 0,
          "width": "100%",
        },
        undefined,
      ]
    }
    useNativeDriver={true}
  >
    <View
      accessibilityLabel="modal-backdrop"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="modal-backdrop"
    >
      <View
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </View>
  </div>
</Modal>
`;

exports[`ModalList should render all items one at a time 3`] = `null`;
